entity-service/src/main/kotlin/com/egm/stellio/entity/web/EntityHandler.kt

```diff
@@ -118,9 +125,11 @@ class EntityHandler(
                 userId
             ).toListOfString()
 
+        val entitiesChunk = entitiesUserCanRead.chunked(limit).getOrNull(page - 1) ?: emptyList()
```

Comment:
this has a major performance problem. if there are 1 million results for a query, you'll retrieve them all from the DB to finally return the 30 results asked by the user.

in a general way, pagination is performed in the database, not in memory once you retrieved all the possible results.

of course, rights management makes it a bit more difficult to do. can you think of a way to handle it?

Reply:
Done